### PR TITLE
☂️ Umbrella Gradle build for gradle-based tutorials

### PR DESCRIPTION
**Pull request by gAmUssA**
Build scans for `build.gradle` projects and runs build on it. 

~No need to add new tutorial manually to `settings.gradle` - gradle will search for `build.gradle` in directories (`templates` dir is ignored).~

**UPD:** changed logic a bit - project with tutorial needs to be added to `setting.gradle`.
Dynamic projects don't work with dependabot. 

CleanShot 2020-07-31 at 01 53 28@2x

Super useful if need to run java only part of the tutorials. It DOESN'T replace harness run.
Also super useful when importing to IDE - all projects will be available for search and refactoring and such.

CleanShot 2020-07-31 at 01 47 38@2x

Previously one had to open individual project manually. 
Also enforcing checks on usage of deprecated APIs - build will fail (I fixed many tests that use old TTD syntax.)

### CODE REVIEW

**Comment by gAmUssA** [comment]
and build passed 🥳

File: _includes/tutorials/connect-add-key-to-source/kstreams/code/tutorial-steps/dev/outputs/actual-input-events.log [hunk 1]
```diff
@@ -1,5 +1,5 @@
-[2019-12-20 20:48:58,029] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
-[2019-12-20 20:48:59,045] INFO ConsumerConfig values: 
+[2020-07-31 05:47:14,405] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
```
[the remark this answers is not in the document]
Comment (gAmUssA) [reply _includes/tutorials/connect-add-key-to-source/kstreams/code/tutorial-steps/dev/outputs/actual-input-events.log:1]: makes sense. I don't remember a reason why we put them under source control in the first place

**Comment by rspurgeon** [comment]
@gAmUssA should I be able to run `./gradlew build` from the root now?  I assume that was the idea... but I get this:

```
07:34:06 in kafka-tutorials on  feature/umbrella-gradle-project
[I] ➜ ./gradlew build
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/rspurgeon/dev/confluentinc/kafka-tutorials/settings.gradle' line: 87

* What went wrong:
A problem occurred evaluating settings 'kafka-tutorials'.
> Tutorial _includes:tutorials:connect-sink-elasticsearch:kstreams:code was not included into the build. Check settings.gradle for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
```

**Comment by gAmUssA** [comment]
@rspurgeon it looks like you have this project on your machine but not in master

**Comment by rspurgeon** [comment]
> @rspurgeon it looks like you have this project on your machine but not in master

Sorry, I don't follow.  I pulled the latest from this PR, my local master is up to date (not sure how this should affect this PR unless it needs to be rebased).

```
09:20:36 in kafka-tutorials on  master took 2s
[I] ➜ gh pr checkout 492
Switched to branch 'feature/umbrella-gradle-project'
Your branch is up to date with 'origin/feature/umbrella-gradle-project'.
Already up to date.


A new release of gh is available: 0.11.0 → v0.11.1
https://github.com/cli/cli/releases/tag/v0.11.1


09:20:41 in kafka-tutorials on  feature/umbrella-gradle-project took 2s
[I] ➜ git rev-parse --abbrev-ref HEAD
feature/umbrella-gradle-project

09:20:51 in kafka-tutorials on  feature/umbrella-gradle-project
[I] ➜ ./gradlew build

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/rspurgeon/dev/confluentinc/kafka-tutorials/settings.gradle' line: 87

* What went wrong:
A problem occurred evaluating settings 'kafka-tutorials'.
> Tutorial _includes:tutorials:connect-sink-elasticsearch:kstreams:code was not included into the build. Check settings.gradle for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 665ms
```

**Comment by gAmUssA** [comment]
@rspurgeon 

```
~/projects/confluent/kafka-tutorials master*
❯ g pl upstream master
From github.com:confluentinc/kafka-tutorials
 * branch              master     -> FETCH_HEAD
Successfully rebased and updated refs/heads/master.

~/projects/confluent/kafka-tutorials master* ⇣⇡
❯ ll _includes/tutorials | grep connect
drwxr-xr-x 5 viktor staff 160 May 27 12:18 connect-add-key-to-source
drwxr-xr-x 3 viktor staff  96 Jul 31 16:07 kafka-connect-datagen-local
```

There is no `connect-sink-elasticsearch` tutorial in master branch as of now.

**Comment by rspurgeon** [comment]
@gAmUssA I may have some abandoned local changes, when I have time I'll git clean and try again

**Comment by gAmUssA** [comment]
Build passes but fails in ksql tutorial (unrelated to this PR)

**Pull request merged**